### DISCUSSION TITLE
lspci command not found in the Jetson NX yocto image built using Ridgerun

### DISCUSSION THREAD

**Discussion by jkdtgit** [OE4T/meta-tegra#921] [General]
Dear Team,
I have referred 
https://developer.ridgerun.com/wiki/index.php?title=Yocto_Support_for_NVIDIA_Jetson_Platforms_-_Setting_up_Yocto
built the image and flashed. 
1. But I dont see lspci command is there in the image.
2. What is the difference between the Ridgerun project and https://github.com/OE4T/tegra-demo-distro git?
Did you customize the project after taking from OE4T?

Thanks.
JK

**Comment by ichergui** [comment]
Hello @jkdtgit 
1. try the following command `$ /usr/sbin/lspci`
2. I don't know exactly the difference. I think that `Ridgerun` is using the same repo `meta-tegra` from OE4T

**Reply by madisongh** [reply]
Questions about RidgeRun's wiki pages are probably best directed at RidgeRun.

As to why you might see `lspci` in one image but not another, here's a general answer:

`lspci` is provided by the `pciutils` package, and in OE-Core, that package is grouped into `packagegroup-base-pci`.  That group is pulled in by `packagegroup-base` via RDEPENDS based on the `pci` feature being enabled in both DISTRO_FEATURES and MACHINE_FEATURES. All of the Jetson machine configurations have that enabled in MACHINE_FEATURES.  Our demo distro also enables PCI support in DISTRO_FEATURES.

So you might not see it show up in an image if you're building a distro that does not enable PCI features (although I'm pretty sure `poky` does), or if you're building an image (such as `core-image-minimal`, perhaps) that does not pull in `packagegroup-base`.  You can also configure build to exclude it, but I suspect that's not the case here.

Or, since @ichergui mentioned that the command is in `/usr/sbin`, it could just be that the package *has* been included, but you don't have that directory on your `PATH`.

**Reply by jkdtgit** [reply]
@ichergui - There is no lspci command under /usr/sbin.
@madisongh I am using core-image-sato-dev image.
Is pci features supported in this image?

or I have to enable in DISTRO_FEATURES and MACHINE_FEATURES and build the entire project or only that particular recipe? 
Basically, can I reduce the build time?

Thanks.

**Reply by madisongh** [reply]
> I am using core-image-sato-dev image.
Is pci features supported in this image?

As far as I know, yes, but I don't use the poky reference distro very much.  I suggest learning more about how the Yocto Project tools and their reference distro work, and how to debug problems like this for your builds; documentation and training links are available at [their web site](https://www.yocotproject.org).
